File: URI/1 - INICIANTE/Python/1118 - VariasNotasComValidacao.py
z = 0
w = 0
nota = nota2 = 0
while z == 0:
    w = 0
    while w == 0:
        nota = float(input())
        if nota >= 0 and nota <= 10:
            w = 1
        else:
            print('nota invalida')
    w = 0
    while w == 0:
        nota2 = float(input())
        if nota2 >= 0 and nota2 <= 10:
            w = 1
        else:
            print('nota invalida')
    print('media = {:.2f}'.format((nota+nota2)/2))
    w = 0
    while w == 0:
        print('novo calculo (1-sim 2-nao)')
        op = int(input())
        if op == 1 or op == 2:
            if op == 2:
                w = 1
                z = 1
                break
            w = 1

File: URI/1 - INICIANTE/C/1118 - VariasNotasComValidacao.c
#include <stdio.h>

int main()
{
	int w = 0, z = 0, op;
	float nota1, nota2;
	
	while(z == 0)
	{
		w = 0;
		while(w == 0)
		{
			scanf(" %f", &nota1);
			if(nota1 >= 0 && nota1 <= 10)
			{
				w = 1;
			}else {
				printf("nota invalida\n");
			}
		}
		w = 0;
		while(w == 0)
		{
			scanf(" %f", &nota2);
			if(nota2 >= 0 && nota2 <= 10)
			{
				w = 1;
			}else {
				printf("nota invalida\n");
			}
		}
		printf("media = %.2f\n",(nota1+nota2)/2);
		w = 0;
		while(w == 0)
		{
			printf("novo calculo (1-sim 2-nao)\n");
			scanf(" %d", &op);
			if(op == 1 || op == 2)
			{
				if(op == 2)
				{
					w =1;
					z =1;
					break;
				}
				w = 1;
			}
		}
	}

	
	return 0;
}
